packages/core/src/internal/newTask.js

```diff
@@ -114,6 +114,7 @@ export default function newTask(env, mainTask, parentContext, parentEffectId, me
     // methods
     cancel,
     cont,
+    digestEffect,
```

Comment:
this is somewhat leaky (but the problem is not exclusive to the digestEffect) - because we give those tasks to users and things like - `digestEffect` should never be called by them (except in effect runners)

We don't necessarily have to fix it, but its a pity nevertheless. Not sure how we should split those in public and kinda protected interfaces

Reply:
Oh, I forgot that the task can be accessed by the users, adding too much fields/methods to the task is a bad idea.